src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -721,7 +722,13 @@ private static TypeSymbol GetNamedPropertyType(ObjectType baseType, IPositionabl
             {
                 if (declaredProperty.Flags.HasFlag(TypePropertyFlags.WriteOnly))
                 {
-                    return new ErrorTypeSymbol(DiagnosticBuilder.ForPosition(propertyExpressionPositionable).WriteOnlyProperty(baseType, propertyName));
+                    var writeOnlyDiagnostic = DiagnosticBuilder.ForPosition(propertyExpressionPositionable).WriteOnlyProperty(TypeValidator.ShouldWarn(baseType), baseType, propertyName);
+                    diagnostics.Add(writeOnlyDiagnostic);
+
+                    if (writeOnlyDiagnostic.Level == DiagnosticLevel.Error)
+                    {
+                        return new ErrorTypeSymbol(Enumerable.Empty<ErrorDiagnostic>());
```

Comment:
>ErrorTypeSymbol [](start = 35, length = 15)

not for this pr - I'm guessing at some point we won't need the ability to have diagnostics on the error type anymore?

Reply:
Yeah I agree. I was wanting to finish the refactor to remove `ErrorDiagnostic` entirely and just use `Diagnostic` across the board which I think would mandate this, but decided against including it in this PR because it's a pretty large set of changes.